### DISCUSSION TITLE
HTTPS to HTTP downgrade isn't working

### DISCUSSION THREAD

**Discussion by mkarne1** [mitmproxy/mitmproxy#6076] [General]
#### Problem Description
I want to downgrade HTTPS to HTTP using [sslstrip.py](https://github.com/mitmproxy/mitmproxy/blob/0b89aede3817831394e6473d82eb04ce572ef621/examples/contrib/sslstrip.py) from @mhils. But no matter which site I'm trying, I get the HTTPS version. 

------------------

#### Steps to reproduce the behavior:
1. Terminal 1: `root@kali: $ettercap -Tq -M arp:remote -i eth0 -S ///`
2. Terminal 2: `root@kali: $./mitmdump -s sslstrip.py`
3. Terminal 3: `root@kali: $iptables -t nat -A PREROUTING -p tcp --destination-port 80 -j REDIRECT --to-port 8080`
4. Open any other machine that's on the same network (Windows 10 machine in my case) and try to open some webpages that use HTTPS (not HSTS, just HTTPS(

Exepected behaviour: The site gets downgraded to HTTP
Actual behaviour: I still get the HTTPS version 

------------------

#### System Information
Paste the output of "mitmproxy --version" here.

Mitmproxy: 9.0.1 binary
Python:    3.11.0
OpenSSL:   OpenSSL 3.0.7 1 Nov 2022
Platform:  Linux-6.1.0-kali7-amd64-x86_64-with-glibc2.36

------------------

#### Used machines 
1. Host OS: macOS BigSur
2. VMWare Fusion with Kali Linux 6.1.0
3. VMWare Fusion with Windows 10 

(Both VM's are using NAT so the NIC for kali is eth0)

------------------

Please let me know if I'm making any stupid mistake and/or how I can fix this issue. 

Thanks in advance!

**Comment by mhils** [comment]
The examples in contrib/ are explicitly without support, so I'm moving this to discussions.

**Comment by Prinzhorn** [comment]
> But no matter which site I'm trying, I get the HTTPS version.

It's possible that every site you've tried is in https://hstspreload.org/ and browsers ship hard coded HSTS.
